tests/unit/test_gbq.py

```diff
@@ -100,6 +118,45 @@ def test_to_gbq_with_no_project_id_given_should_fail(monkeypatch):
     assert "Could not determine project ID" in str(exception)
 
 
+def test_to_gbq_read_projectid_from_service_account_credentials(
+    min_bq_version
+):
+    import pkg_resources
+
+    credentials, _ = mock_get_service_account_credentials()
+    pandas_version = pkg_resources.parse_version("0.24.0")
+    with mock.patch(
+        "pkg_resources.Distribution.parsed_version",
+        new_callable=mock.PropertyMock,
+    ) as mock_version:
+        mock_version.side_effect = [min_bq_version, pandas_version]
+        try:
+            gbq.to_gbq(
+                DataFrame([[1]]), "dataset.tablename", credentials=credentials
+            )
+        except gbq.TableCreationError:
+            pass
```

Comment:
Another nit. I'd like the test to `assert` something. Could you follow the `test_read_gbq` example instead of `to_gbq` and assert that it returns a DataFrame?